tmp/lanelet2_extension/lib/utilities.cpp

```diff
@@ -501,12 +501,28 @@ lanelet::ConstLanelets getExpandedLanelets(
 }
 
 void overwriteLaneletsCenterline(
-  lanelet::LaneletMapPtr lanelet_map, const double resolution, const bool force_overwrite)
+  lanelet::LaneletMapPtr lanelet_map, const double resolution, const bool use_waypoints,
+  const bool force_overwrite)
 {
   for (auto & lanelet_obj : lanelet_map->laneletLayer) {
-    if (force_overwrite || !lanelet_obj.hasCustomCenterline()) {
+    if (force_overwrite) {
       const auto fine_center_line = generateFineCenterline(lanelet_obj, resolution);
       lanelet_obj.setCenterline(fine_center_line);
+    } else {
+      if (use_waypoints) {
+        if (lanelet_obj.hasCustomCenterline()) {
+          const auto & centerline = lanelet_obj.centerline();
+          lanelet_obj.setAttribute("waypoints", centerline.id());
+        }
+
+        const auto fine_center_line = generateFineCenterline(lanelet_obj, resolution);
+        lanelet_obj.setCenterline(fine_center_line);
```

Comment:
I'm a bit concerned that the function start to exceed what it was originally designed for. 
Is it necessary to move the customized center line to `waypoints` at runtime?

```
/**
 * @brief  Apply a patch for centerline because the original implementation
 * doesn't have enough quality
 */
```